### ISSUE TITLE
Custom songs don't work

### ISSUE THREAD

**Issue by UltraBlackLinux** [beat-feet/beat-feet#109]
Hey there,
I just wanted to add a custom song to the game, but I can't seem to play it. There is no button or anything to play it, even though I've made sure to have the path correct - What am I doing wrong?
Thanks!

**Comment by pserwylo** [comment]
Hi @UltraBlackLinux, thanks for the report. Could I ask what you see on the screen after you try to launch a custom level? Do you see the loading screen? Or something else? If the loading screen, how long does it stay there?

There are performance problems with loading custom songs which makes them take a long time, but also if they take too long they tend to crash eventually (if so, then this is likely a duplicate of #85).

**Comment by UltraBlackLinux** [comment]
> Hi @UltraBlackLinux, thanks for the report. Could I ask what you see on the screen after you try to launch a custom level? Do you see the loading screen? Or something else? If the loading screen, how long does it stay there?
> 
> There are performance problems with loading custom songs which makes them take a long time, but also if they take too long they tend to crash eventually (if so, then this is likely a duplicate of #85).

The custom levels just don't show up at all

**Comment by sebastien46** [comment]
Could we have an update on the situation ? I have the same problem : copied mp3 to path, went to levels menu in the game, it doesn't show up 😑

**Comment by pserwylo** [comment]
Yeah, sorry for the radio silence, and I appreciate you coming by to ask about it @sebastien46.

I've been spending some time on other aspects of this game lately (namely adding new levels). Once that is out, then I'll pivot back to this, as it there are actually several things I dislike about the current implementation.

First and foremost, it should not try to load the whole song in memory (#85). However, it should also be much easier to find and work with your own files. The reason I want to wait for my current work to go out first is because it introduces the notion of "Worlds" (think Super Mario Brothers how levels are grouped into worlds). To add your own songs, I want to not just improve the situation on a technical level to address #85, but also on a user experience level so that you can do something like:
* "Add new world"
* "Select an mp3"
* ... browse your system to find the file ...
* Show a series of custom songs together on the screen for you to choose from.

This will take time, so I appreciate your patience, but rest assured it is something high on my list of priorities (but below spending time with the family and a full time job :wink:)
